config/ci/rbac/rbac_patch.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  creationTimestamp: null
+  name: manager-event-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: event-role
+subjects:
+- kind: ServiceAccount
+  name: default
+  namespace: system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  creationTimestamp: null
+  name: event-role
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - events
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - patch
+  - delete
```

Comment:
Instead of a patch, this should probably be set using a kubebuilder annotation.

I don't see any reason why we shouldn't add it to all of the controllers by default assuming that we can scope down the verbs (we probably shouldn't allow the controller to delete, patch or update existing events).

Reply:
@detiber i am good with that, although the current manager that running machineSet controller does not generate event, i see a potential usage, and i limited the events scope to: ```get, list, watch, create```